public/apps/audit/components/main/main.js

```diff
@@ -0,0 +1,340 @@
+import {
+  AUDIT_API,
+  CONFIG_LABELS,
+  SETTING_GROUPS,
+  RESPONSE_MESSAGES,
+  CALLOUT_MESSAGES,
+} from './config';
+import ContentPanel from './ContentPanel';
+import DisplaySettingGroup from './DisplaySettingGroup';
+import EditSettingGroup from './EditSettingGroup';
+import { generateReadonlyPaths } from './utils';
+
+export function Main(props) {
+  const [config, setConfig] = useState(null);
+  const [editConfig, setEditConfig] = useState(null);
+  const [readonly, setReadonly] = useState([]);
+  const [showAllSettings, setShowAllSettings] = useState(true);
+  const [showConfigureAudit, setShowConfigureAudit] = useState(false);
+  const [showConfigureCompliance, setShowConfigureCompliance] = useState(false);
+  const [showError, setShowError] = useState(false);
+  const [invalidSettings, setInvalidSettings] = useState([]);
+
+  useEffect(() => {
+    fetchConfig();
+  }, [props.httpClient]);
+
+  const handleChange = (setting, val) => {
+    const editedConfig = set(cloneDeep(editConfig), setting.path, val);
+    setEditConfig(editedConfig);
+  };
+
+  const handleChangeWithSave = (setting, val) => {
+    const editedConfig = set(cloneDeep(editConfig), setting.path, val);
+    saveConfig(editedConfig, false);
+  };
+
+  const handleInvalid = (key, error) => {
+    const invalid = pull([...invalidSettings], key);
+    if (error) {
+      invalid.push(key);
+    }
+    setInvalidSettings(invalid);
+  };
+
+  const toggleDisplay = (
+    show_all_settings = true,
+    show_configure_audit = false,
+    show_configure_compliance = false
+  ) => {
+    setShowAllSettings(show_all_settings);
+    setShowConfigureAudit(show_configure_audit);
+    setShowConfigureCompliance(show_configure_compliance);
+    window.scrollTo({ top: 0 });
+  };
+
+  const cancel = () => {
+    toggleDisplay();
+    setEditConfig(config);
+    setInvalidSettings([]);
+  };
+
+  const saveConfig = (payload, showToast = true) => {
+    const { httpClient } = props;
+    httpClient
+      .post(AUDIT_API.PUT, payload)
+      .then(() => {
+        if (showToast) {
+          toastNotifications.addSuccess(RESPONSE_MESSAGES.UPDATE_SUCCESS);
+        }
+        toggleDisplay();
+        fetchConfig();
+      })
+      .catch(() => {
+        toastNotifications.addDanger(RESPONSE_MESSAGES.UPDATE_FAILURE);
+      });
+  };
+
+  const fetchConfig = () => {
+    const { httpClient } = props;
+    httpClient
+      .get(AUDIT_API.GET)
+      .then(resp => {
+        const responseConfig = resp.data.data.config;
+        const readonly = generateReadonlyPaths(resp.data.data._readonly);
+        setConfig(responseConfig);
+        setEditConfig(responseConfig);
+        setReadonly(readonly);
+        setShowError(false);
+      })
+      .catch(() => {
+        setShowError(true);
+      });
+  };
+
+  const renderCallout = messages => {
+    return (
+      <>
+        <EuiCallOut title="Warning" color="warning" iconType="alert">
```

Comment:
Just want to call out that the mockup of the call out has not icon.

Reply:
Removed the alert icon